### PR TITLE
install istanbul and add coverage scripts also gitignore the coverage…

### PR DESCRIPTION
**Pull request by eanplatter**
… folder

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,6 +6,8 @@
   "main": "dist/TapFlux.js",
   "scripts": {
     "test": "mocha",
+    "test-cov": "istanbul cover _mocha -- -R spec test",
+    "check-cov": "istanbul check-coverage --statements 100 --functions 100 --branches 100 --lines 100",
     "test-watch": "mocha --watch",
```
Comment (eanplatter) [inline package.json]: `test-cov` will run the cover function and create the coverage folder. `check-cov` will throw an error of coverage is too low.

File: .gitignore [hunk 2]
```diff
@@ -3,6 +3,6 @@
 .DS_STORE
 .idea
 npm-debug.log
-
+coverage
```
Comment (eanplatter) [inline .gitignore]: wanna make sure we're not pushing that coverage folder, but within it there is a folder that has an `index.html` file. Throw it in the browser and get a sweet gui for coverage.

File: package.json (hunk 1 above)
Comment (levithomason) [inline package.json]: Sweet deal, how about some `README.md` notes for these?

**Comment by levithomason** [comment]
Would really like to see some output in the build step.

The workflow I'd like us to get to is one where watchers make everything happen that should and if something is wrong or yucky, it doesn't happen.  So, if there is a lint issue or missing coverage, you will immediately know and can fix it.

I see two potential gulp plugins for this:

1. https://github.com/SBoudrias/gulp-istanbul (officially acknowledged by gulp plugins)
1. https://github.com/iainjmitchell/gulp-istanbul-enforcer (not acknowledged)

There may be more, just did a quick scan.

**Comment by TomFrost** [comment]
@levithomason Yep, those are solid. Here's how I did it in Jexl: https://github.com/TechnologyAdvice/Jexl/blob/master/gulpfile.js#L35

Putting a watch on that would be handy. Especially if `gulp-eslint` is added in the aslant branch and both of these are added to a parent task.

**Comment by levithomason** [comment]
Nice. I love to have a constant build stream going.  With `gulp-plumber` you can set these up to still yell and not stop the build loop.  Results in endless testing, linting, and building as you code (...with auto save!).

OH MAN!  My head is gonna explode with glorious build hampsters :hamster:  :hamster:  :hamster:

**Comment by eanplatter** [comment]
☆*:.｡. o(≧▽≦)o .｡.:*☆

**Comment by levithomason** [comment]
:+1:

**Pull request merged**